### PR TITLE
errlist: Fix issue with no errors

### PR DESCRIPTION
**Pull request by yauhen-l**
The 2nd approach.
Fix issue with no errors.
Add tests on generated validators.

### CODE REVIEW

File: errlist/fielderr.go [hunk 1]
```diff
@@ -4,14 +4,14 @@ import "fmt"
 
 type FieldErr struct {
 	Field string `json:"field"`
-	Err   error  `json:"err"`
+	Err   string `json:"err"`
```
Comment (mkabischev) [inline errlist/fielderr.go]: Why did you change type for `Err`?

File: examples/complicated/validators.go [hunk 2]
```diff
@@ -30,7 +30,10 @@ func (r AliasArray) Validate() error {
 		_ = i
 		_ = x
 	}
-	return &errs
+	if errs.HasErrors() {
```
Comment (mkabischev) [inline examples/complicated/validators.go]: What about create some method like `ErrorOrNil` attached to `ErrList` which will do this inside:
```go
func (e *ErrList) ErrorOrNil() error {
    if e.HasErrors() {
        return e
    }
    
    return nil
}```

Name `ErrorOrNil` is just proposal, I want to discuss the idea. `pkg/errors` has very similar logic inside `wrap` method.

File: examples/simple/validators_test.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
+package simple
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_User_Validate(t *testing.T) {
```
Comment (mkabischev) [inline examples/simple/validators_test.go]: Thanks for tests 👍

File: examples/complicated/validators.go (hunk 2 above)
Comment (yauhen-l) [inline examples/complicated/validators.go]: I did it in the first approach.
 did it wrong:
`func (e *ErrList) ErrorOrNil() *ErrList`
I want to evidently return `ErrList` from `Validate` method, so in consumer I can cast to ErrList.
Will try your suggestion.

File: errlist/fielderr.go (hunk 1 above)
Comment (yauhen-l) [inline errlist/fielderr.go]: To not override `Marshal` :)
Because `error` is marshaled without call of `Error()` method.

File: errlist/fielderr.go [hunk 4]
```diff
@@ -1,17 +1,28 @@
 package errlist
 
-import "fmt"
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+)
+
+const UnknownField = "unknown"
```
Comment (mkabischev) [inline errlist/fielderr.go]: Why this const is exported?

File: errlist/errlist.go [hunk 5]
```diff
@@ -2,19 +2,20 @@ package errlist
 
 import (
 	"bytes"
-	"fmt"
+
+	"github.com/pkg/errors"
 )
 
-type ErrList []error
+type ErrList []FieldErr
```
Comment (mkabischev) [inline errlist/errlist.go]: Really I don\`t like name `errlist.ErrList`. Why not just `errlist.List`. Also why `AddFieldErr` not just `Add`, `Push` or `Append`. I think it\`s better to use some term more related to list.

File: errlist/fielderr.go (hunk 4 above)
Comment (yauhen-l) [inline errlist/fielderr.go]: `genval` package returns `[]FieldErr`.
If `genval` cannot determine field, consumer can handle `unkown` field somehow.

File: errlist/errlist.go (hunk 5 above)
Comment (yauhen-l) [inline errlist/errlist.go]: Discussed.

**Comment by mkabischev** [comment]
Also what about nested errors? Imagine I have struct with `ID` and some nested struct also with `ID` field. In current implementation it\`s impossible to understand which field isn\`t correct.

**Comment by yauhen-l** [comment]
You're right. Currently slice of errors will be plain. We discussed with @mkorolyov  to keep process iterative and go with current solution. Then improve it.
